doc/source/structure.rst

```diff
@@ -0,0 +1,108 @@
+
+* ``/core``
+* ``/dem``
+* ``/fem-dem``
+* ``/rpt``
+* ``/solvers``
+
+The content of Lethe is subdivided into modules that are generally independent of one another. This encapsulates furthermore the content of the software. The core module contains the core functionality of Lethe, whereas the solvers module contains the applications that are related to the finite element solution of multiphysics problems. The dependency between the module can only be unidirectional. For example, the solvers module requires the core module, but the core module could not require the solvers module. 
+
+
+The tests folder
+-----------------
+
+The ``/tests`` folder houses the unit tests for all of the modules of Lethe. The unit tests are different from the application's tests. Their scope is only to verify that a single class or functions behave in the expected manner. They are used to test individual components of the Lethe and not complete applications. Hence, these tests are grouped by modules.
+
+The prototypes folder
+---------------------
+
+The ``/prototypes`` folder houses temporary applications in Lethe. These applications are prototypes that are very close to the steps of deal.II. They are used to test novel development before they are fully integrated within Lethe. The prototypes are not expected to be stable.
+
+The prototypes folder contains a template folder which contains a virgin prototype. Each prototype should be implemented within their own folder. When implementing a new prototype, the following should be done:
+
+* Copy the template folder to a new folder within the prototypes folder that bears the name of what you want to develop
+* Within the new folder, modify the ``CMakeLists.txt`` file to ensure that the name of the application has the name you want it to have. This is achieved by modifying the ``SET(TARGET "protype_template")`` line to ``SET(TARGET "my_project_name")``
+* Modify the name of the .cc files so that it corresponds to the name of your prototype. All .cc and .h files will be detected automatically by the ``CMakeLists.txt`` GLOB instructions
+* Add the new folder to the list of prototypes to be taken into account by modifying the ``prototypes/CMakeLists.txt`` and adding a ``ADD_SUBDIRECTORY(my_prototype)``
+
+
+The contrib folder
+-------------------
+
+The ``/contrib`` folder houses various scripts that help for the maintenance of Lethe. It notably contains the ``contrib/utilities/indent-all`` script which can be used to automatically indent the entire Lethe source code to ensure that it adheres to the clang-format rules contained in the ``/contrib`` folder.
+
+The doc folder
+--------------
+
+The ``\doc`` folder contains the source files of this documentation page of Lethe. To contribute or compile the documentation on your own machine follow the instructions of the *contributing* tab.
```

Comment:
Do you know if it is possible to hyperlink to the contributing time from inside the documentation? a bit like how we can do references in a single file?
If it's not possible, let's not worry about it, but if @technophil98 you have an idea ? :).

Reply:
Yes, it is possible! Just pushed a commit using the hyperlink :)